grails-app/services/de/iteratec/osm/result/ApplicationDashboardService.groovy

```diff
@@ -173,4 +178,41 @@ class ApplicationDashboardService {
         jobGroup.save(failOnError: true, flush: true)
         return csiConfiguration.id
     }
+
+    ApplicationCsiDto getCsiValuesAndErrorsForJobGroup(JobGroup jobGroup) {
+        ApplicationCsiDto applicationCsiListDto = new ApplicationCsiDto()
```

Comment:
If a JobGroup does not have a CsiConfig none of the following operations need to be performed. You are able to return this straight at the start of this method